File: searching/binary_search.py
import math

array = [1, 10, 3, 5, 2, 6, 4, 11, 7];

def binarySearch(arr, el):
    start = -1
    end = len(arr)
    arr.sort()
    while end - start > 1:
        mid = math.floor((start + end) / 2)
        if arr[mid] == el: 
            return mid
        if arr[mid] > el:
            end = mid
        else:
            start = mid

    return -1

print(binarySearch(array, 12));

File: searching/binary_search.js
let array = [1, 3, 4, 2, 8, 6, 4, 6, 12, 9];

function binarySearch(arr, el) {
    let start = -1;
    let end = arr.length;
    arr.sort((a, b) => a - b);

    while(end - start > 1) {
        let mid = Math.floor((start + end) / 2);
        if (arr[mid] === el) return mid;

        if (arr[mid] > el) {
            end = mid;
        }
        else {
            start = mid;
        }
    }
    return -1;
}

console.log(binarySearch(array, 12));